### PR TITLE
Guarin lig 836 add tests to dino tricks pr

### PR DESCRIPTION
**Pull request by guarin**
* Add tests for dino tricks pr #731 
* Fix freeze last layer implementation
* Improve dino loss tests to make sure that really everything works as expected (it does)
* Cleanup examples

### CODE REVIEW

**Review by MalteEbner — COMMENTED** [review]
Great that you added the unittests!

File: tests/models/test_ProjectionHeads.py [hunk 1]
```diff
@@ -116,3 +117,42 @@ def test_dino_projection_head(self, device="cpu", seed=0):
...
+                    optimizer.zero_grad()
+                    loss.backward()
+                    head.cancel_last_layer_gradients(current_epoch=epoch)
+                    optimizer.step()
+                    params = head.last_layer.parameters()
+                    # Verify that weights have (not) changed depending on epoch.
+                    for param, init_data in zip(params, initial_data):
+                        if param.requires_grad:
+                            are_same = torch.allclose(param.data, init_data)
+                            if epoch >= 2:
```
[the remark this answers is not in the document]
Comment (guarin) [reply tests/models/test_ProjectionHeads.py]: Done :)

**Review by MalteEbner — APPROVED** [review]
LGTM
